src/custom/components/swap/EthWethWrap/index.tsx

```diff
@@ -113,160 +67,448 @@ const ButtonWrapper = styled.div`
   margin-top: 8px;
 `
 
-const ErrorMessage = ({ error }: { error: Error }) => (
-  <ErrorWrapper>
-    <strong>{error.message}</strong>
-  </ErrorWrapper>
-)
-
-const WarningLabel = ({ children }: { children?: ReactNode }) => (
-  <WarningWrapper>
-    <AlertTriangle size={25} />
-    <div>{children}</div>
-  </WarningWrapper>
-)
+const MODAL_CLOSE_DELAY = 1000 // 1s
 
 export interface Props {
   account?: string
   native: Currency
-  nativeInput?: CurrencyAmount<Currency>
   wrapped: Token & { logoURI: string }
-  wrapCallback: () => Promise<TransactionResponse>
+  nativeInput?: CurrencyAmount<Currency>
+  isNativeIn: boolean
+  isNativeOut: boolean
+  isWrappedIn: boolean
+  isWrappedOut: boolean
+
+  // state
+  needsApproval: boolean
+  needsWrap: boolean
+
+  // cbs
+  onDismiss: () => void
+  approveCallback: (params?: { useModals: boolean }) => Promise<TransactionResponse | undefined>
+  wrapCallback: ((params?: { useModals: boolean }) => Promise<TransactionResponse | undefined>) | undefined
+  swapCallback: (showConfirmModal?: boolean) => void
 }
 
-export default function EthWethWrap({ account, native, nativeInput, wrapped, wrapCallback }: Props) {
+export default function EthWethWrap({
+  account,
+  native,
+  wrapped,
+  nativeInput,
+  isNativeIn,
```

Comment:
(just a thought: probably not actionable)
To determine your actions, you are handing over a lot of bool variables(nativeInput,isNativeIn,isNativeOut,isWrappedIn,isWrappedOut,) to the EthWethWrap function. 
Maybe it would have been nicer to hand over a flow type enum: 
```
enum ETHWethWrapType {
WrapETH,
UnwrapETH,
WrapETHAndTrade
}
'
and having a helper function that calculates the enum ETHWethWrapType from the bool variables. This could increase the readability of the code. 
But it seems that your way is much easier and also more the normal FE design pattern.

Reply:
I 100% agree that there are too many prop bool flags being sent thru and it could be optimised. I'll take a look at it and consider your suggestion, I like it

Reply:
done [7325ff9](https://github.com/cowprotocol/cowswap/pull/889/commits/7325ff9b3cd419881b83fe0bba79ff5ac0c20809)